src/session/basic_group.rs

```diff
@@ -109,4 +110,22 @@ impl BasicGroup {
         self_.member_count.set(member_count);
         self.notify("member-count");
     }
+
+    pub fn member_count_expression(&self) -> gtk::Expression {
```

Comment:
I see 2 issues with this function.

The first issue is that it won't “fire” when the `member-count` is changed by an update from tdlib as you do not bind it to the property.

The second is somehow related to the first: The function breaks the scheme we expose expressions. For example, take a look at `User::first_name_expression()`: A user has a property `first-name` and we provide an Expression just for that property. You are already adding too much context here by returning a ClosureExpression that resolved to a string "n members" or "1 member". You should just expose the property `member-count` here and provide the context later.

Reply:
> You should just expose the property `member-count` here and provide the context later, where it is needed.

To be clear, I think it would be enough if you change the name of the existing function to `fn formatted_member_count_expression` and in that use the `member_count_expression()` function.

Reply:
> The first issue is that it won't “fire” when the member-count is changed by an update from tdlib as you do not bind it to that property. 
How can I fix it?

Reply:
I've created function called member_count_expression() and called it inside formatted_member_count_expression() but doesn't seem to work

Reply:
This is how I did it for the BasicGroup
```rust
    pub fn member_count_expression(basic_group_expression: &gtk::Expression) -> gtk::Expression {
        gtk::PropertyExpression::new(
            Self::static_type(),
            Some(basic_group_expression),
            "member-count",
        )
        .upcast()
    }

    pub fn formatted_member_count_expression(
        basic_group_expression: &gtk::Expression,
    ) -> gtk::Expression {
        let member_count_expression = Self::member_count_expression(basic_group_expression);
        gtk::ClosureExpression::new(
            |args| {
                let member_count = args[1].get::<i32>().unwrap();
                if member_count > 1 {
                    // Translators: <NUM> group members
                    gettext!("{} members", member_count)
                } else {
                    gettext("1 member")
                }
            },
            &[member_count_expression],
        )
        .upcast()
    }
```
And then the relevant part in `chat_history.rs`
```rust
       // ...
        ChatType::BasicGroup(data) => {
            BasicGroup::member_count_expression(gtk::ConstantExpression::new(data).upcast_ref())
        }
       // ...
```

But it will probably change with the updated gtk-rs 0.4 crate. Haven't had a deeper look.
How would you like us to proceed? Do you want that we first resolve all issues and then do the rebase or the other way arround?

Reply:
In fact that doesn't solve the issue at 100% for me
![image](https://user-images.githubusercontent.com/58047802/149678235-30a7e389-48b3-4ba4-8978-fe49a6ba4943.png)
![image](https://user-images.githubusercontent.com/58047802/149678238-56437779-7d94-4ef3-ac5e-e91fc6871186.png)

The counter is updating, but shows smaller number for some chats. I susspect tdlib/tl bug, basically tdlib is limitted to send updates to only one device/client at the time. That may result missmatched data like that:
![image](https://user-images.githubusercontent.com/58047802/149678432-223a78e3-e9d9-438a-8dda-9a27a6531a1e.png)
The offical client is able to handle that error and fix data next time you select chat, while data in the tdgrand/telegrand stays incorrect forever. But it's also possible, that this error occurred because of my bad code and reomoving cache will resolve it. 

I think it's better to rebase this branch before merging, I'll do it tomorrow

Reply:
> In fact that doesn't solve the issue at 100% for me

Hmm weird, at least from my tests everything seems working. But only tested with two users.

> I susspect tdlib/tl bug, basically tdlib is limitted to send updates to only one device/client at the time.

Where did you find that information?

> I think it's better to rebase this branch before merging, I'll do it tomorrow

Ok.

Reply:
> In fact that doesn't solve the issue at 100% for me

Maybe I have an idea why it is not 100% working. Whenever a chat is set in the `ChatHistory::set_chat` method, you bind the `gtk::Expression`'s, but you never unwatch them. In this way, the old binding is still active and may overwrite the new one.

Reply:
This is still not solved. The previous expression still overwrites the current one
Steps to reproduce:

1. Open a group chat
2. Open another chat, preferable one with a user
3. Delete a user from the group chat that was first opened while still being in the second chat

Now, the subtitle of the user chat has the string `n members` :-P

Reply:
This is an example of how to fix this: https://github.com/melix99/telegrand/blob/350aa44d44d7c9810d19f2885ce216ad8e73bd26/src/session/content/message_row/photo.rs#L60